res/res_pjsip.c

```diff
@@ -1000,9 +1000,11 @@ static pjsip_dialog *create_dialog_uas(const struct ast_sip_endpoint *endpoint,
 
 	contact.ptr = pj_pool_alloc(rdata->tp_info.pool, PJSIP_MAX_URL_SIZE);
 	contact.slen = pj_ansi_snprintf(contact.ptr, PJSIP_MAX_URL_SIZE,
-			"<%s:%s%.*s%s:%d%s%s>",
+			"<%s:%s%s%s%.*s%s:%d%s%s>",
 			uas_use_sips_contact(rdata) ? "sips" : "sip",
 			(type & PJSIP_TRANSPORT_IPV6) ? "[" : "",
+			(!ast_strlen_zero(endpoint->contact_user))?endpoint->contact_user:"",
```

Comment:
Please see the coding guidelines. Whitespace is required between operators such as ? and :

Additionally, you could simplify by using the `S_OR` macro.